src/main/java/appeng/core/sync/packets/PacketMEFluidInventoryUpdate.java

```diff
@@ -0,0 +1,207 @@
+{
+	private static final int UNCOMPRESSED_PACKET_BYTE_LIMIT = 16 * 1024 * 1024;
+	private static final int OPERATION_BYTE_LIMIT = 2 * 1024;
+	private static final int TEMP_BUFFER_SIZE = 1024;
+	private static final int STREAM_MASK = 0xff;
+
+	// input.
+	@Nullable
+	private final List<IAEFluidStack> list;
+	// output...
+	private final byte ref;
+
+	@Nullable
+	private final ByteBuf data;
+	@Nullable
+	private final GZIPOutputStream compressFrame;
+
+	private int writtenBytes = 0;
+	private boolean empty = true;
+
+	// automatic.
+	public PacketMEFluidInventoryUpdate( final ByteBuf stream ) throws IOException
+	{
+		this.data = null;
+		this.compressFrame = null;
+		this.list = new LinkedList<>();
+		this.ref = stream.readByte();
+
+		// int originalBytes = stream.readableBytes();
+
+		final GZIPInputStream gzReader = new GZIPInputStream( new InputStream()
+		{
+			@Override
+			public int read() throws IOException
+			{
+				if( stream.readableBytes() <= 0 )
+				{
+					return -1;
+				}
+
+				return stream.readByte() & STREAM_MASK;
+			}
+		} );
+
+		final ByteBuf uncompressed = Unpooled.buffer( stream.readableBytes() );
+		final byte[] tmp = new byte[TEMP_BUFFER_SIZE];
+		while( gzReader.available() != 0 )
+		{
+			final int bytes = gzReader.read( tmp );
+			if( bytes > 0 )
+			{
+				uncompressed.writeBytes( tmp, 0, bytes );
+			}
+		}
+		gzReader.close();
+
+		// int uncompressedBytes = uncompressed.readableBytes();
+		// AELog.info( "Receiver: " + originalBytes + " -> " + uncompressedBytes );
+
+		while( uncompressed.readableBytes() > 0 )
+		{
+			this.list.add( AEFluidStack.fromPacket( uncompressed ) );
+		}
+
+		this.empty = this.list.isEmpty();
+	}
+
+	// api
+	public PacketMEFluidInventoryUpdate() throws IOException
+	{
+		this( (byte) 0 );
+	}
+
+	// api
+	public PacketMEFluidInventoryUpdate( final byte ref ) throws IOException
+	{
+		this.ref = ref;
+		this.data = Unpooled.buffer( OPERATION_BYTE_LIMIT );
+		this.data.writeInt( this.getPacketID() );
+		this.data.writeByte( this.ref );
+
+		this.compressFrame = new GZIPOutputStream( new OutputStream()
+		{
+			@Override
+			public void write( final int value ) throws IOException
+			{
+				PacketMEFluidInventoryUpdate.this.data.writeByte( value );
+			}
+		} );
+
+		this.list = null;
+	}
+
+	@Override
+	@SideOnly( Side.CLIENT )
+	public void clientPacketData( final INetworkInfo network, final AppEngPacket packet, final EntityPlayer player )
+	{
+		final GuiScreen gs = Minecraft.getMinecraft().currentScreen;
+
+		if( gs instanceof GuiFluidTerminal )
+		{
+			( (GuiFluidTerminal) gs ).postUpdate( this.list );
+		}
+	}
+
+	@Nullable
+	@Override
+	public FMLProxyPacket getProxy()
+	{
+		try
+		{
+			this.compressFrame.close();
+
+			this.configureWrite( this.data );
+			return super.getProxy();
+		}
+		catch( final IOException e )
+		{
+			AELog.debug( e );
+		}
+
+		return null;
+	}
+
+	public void appendFluid( final IAEFluidStack fs ) throws IOException, BufferOverflowException
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.nio.BufferOverflowException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)